frost-rerandomized/src/lib.rs

```diff
@@ -1,97 +1,141 @@
+    frost::{
+        self, compute_lagrange_coefficient,
+        keys::{KeyPackage, PublicKeyPackage, SigningShare, VerifyingShare},
+        Identifier,
+    },
     Ciphersuite, Error, Field, Group, Scalar, VerifyingKey,
 };
 
 // When pulled into `reddsa`, that has its own sibling `rand_core` import.
 // For the time being, we do not re-export this `rand_core`.
 use rand_core::{CryptoRng, RngCore};
 
-/// Performed once by each participant selected for the signing operation.
+// Compute the randomizer share (α^) from the set of `participants` identifiers
+// and the randomizer (α).
+fn compute_randomizer_share<C: Ciphersuite>(
+    participants: &BTreeSet<Identifier<C>>,
+    randomizer: &Scalar<C>,
+) -> Result<Scalar<C>, Error<C>> {
+    let lagrange_sum = participants
+        .iter()
+        .map(|i| compute_lagrange_coefficient(participants, None, *i))
+        .reduce(|acc, e| Ok(acc? + e?))
+        .ok_or(Error::IncorrectNumberOfIdentifiers)?;
+    let randomizer_share =
+        *randomizer * <<C::Group as Group>::Field as Field>::invert(&lagrange_sum?)?;
+    Ok(randomizer_share)
+}
+
+/// Randomize the given [`KeyPackage`] for usage in a rerandomized FROST signing,
+/// using the given [`RandomizedParams`].
 ///
-/// Implements [`sign`] from the spec.
+/// It's recommended to use [`sign`] directly which already handles
+/// the key package randomization.
 ///
-/// Receives the message to be signed and a set of signing commitments and a set
-/// of randomizing commitments to be used in that signing operation, including
-/// that for this participant.
+/// You MUST NOT reuse the randomized key package for more than one signing.
+pub fn randomize_key_package<C: Ciphersuite>(
+    key_package: &KeyPackage<C>,
+    randomized_params: &RandomizedParams<C>,
+) -> Result<KeyPackage<C>, Error<C>> {
+    let verifying_share = key_package.public();
+    let randomized_verifying_share = VerifyingShare::<C>::new(
+        verifying_share.to_element() + randomized_params.randomizer_share_element,
+    );
+
+    let signing_share = key_package.secret_share();
+    let randomized_signing_share =
+        SigningShare::new(signing_share.to_scalar() + randomized_params.randomizer_share);
+
+    let randomized_key_package = KeyPackage::new(
+        *key_package.identifier(),
+        randomized_signing_share,
+        randomized_verifying_share,
+        randomized_params.randomized_verifying_key,
+    );
+    Ok(randomized_key_package)
+}
+
+/// Randomized the given [`PublicKeyPackage`] for usage in a rerandomized FROST
+/// aggregation, using the given [`RandomizedParams`].
 ///
-/// Assumes the participant has already determined which nonce corresponds with
-/// the commitment that was assigned by the coordinator in the SigningPackage.
+/// It's recommended to use [`aggregate`] directly which already handles
+/// the public key package randomization.
+pub fn randomize_public_key_package<C: Ciphersuite>(
```

Comment:
Any reason not to define this as a method on `PublicKeyPackage`, maybe as its own trait `Randomize` or something, and similar for `KeyPackage`? 

```
impl<C> Randomize<C> for PublicKeyPackage<C> {

    fn try_randomize(&self, &RandomizedParams<C>) -> Result<PublicKeyPackage<C>, Error<C>> {
        ...
    }

}
```

Reply:
Great idea! I switched to that in 8480872c63d47d4735559a0d0d02f2bbb14dcb27.

I think we previously agreed to not include rerandomization stuff in `frost-core` even under a flag. But we can revisit that discussion if you want (specially now that it's much less invasive). I don't mind either way.

Reply:
Good point, it's even more tempting to include it in `-core` because the code is simpler

Reply:
Marking this as solved, we can revisit this later